File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionIsNull.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionIsNull extends TransformerFunction {
    public TransformerFunctionIsNull() {
        super();
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var value = context.getJsonElement(null, true);
        return context.getAdapter().isNull(value);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionIsNull.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { isNullOrUndefined } from "../JsonHelpers";

class TransformerFunctionIsNull extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const value = await context.getJsonElement(null);
    return isNullOrUndefined(value);
  }
}

export default TransformerFunctionIsNull;
